azurerm/resource_arm_firewall.go

```diff
@@ -247,7 +257,16 @@ func expandArmFirewallIPConfigurations(d *schema.ResourceData) (*[]network.Azure
 		data := configRaw.(map[string]interface{})
 		name := data["name"].(string)
 		subnetId := data["subnet_id"].(string)
-		intPubID := data["internal_public_ip_address_id"].(string)
+
+		exist := false
```

Comment:
This doesn't work as the properties are empty strings when unset, so exist is always true

```suggestion
		pubID := data["internal_public_ip_address_id"].(string)
		if pubID == "" {
			pubID = data["public_ip_address_id"].(string)
		}

		if pubID == "" {
			return nil, nil, nil, fmt.Errorf("one of `ip_configuration.0.internal_public_ip_address_id` or `ip_configuration.0.public_ip_address_id` must be set")
		}
```